src/vs/workbench/contrib/externalTerminal/browser/externalTerminal.contribution.ts

```diff
@@ -60,12 +60,12 @@ CommandsRegistry.registerCommand({
 						query: resource.query,
 						path: dirname(resource.path)
 					});
-				}).forEach(cwd => {
+				}).forEach(async cwd => {
 					if (opened[cwd.path]) {
 						return;
 					}
 					opened[cwd.path] = true;
-					const instance = integratedTerminalService.createTerminal({ config: { cwd } });
+					const instance = await integratedTerminalService.createTerminal({ config: { cwd } });
 					if (instance && (resources.length === 1 || !resource || cwd.path === resource.path || cwd.path === dirname(resource.path))) {
 						integratedTerminalService.setActiveInstance(instance);
```

Comment:
Changing this to a for const of would be best as I don't believe the async function gets awaited here